appengine-standard-java8/helloworld-launch-dataflow-from-gae/src/main/java/com/example/appengine/java8/HelloAppEngine.java

```diff
@@ -0,0 +1,112 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Properties;
+import java.util.UUID;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.io.IOUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+// With @WebServlet annotation the webapp/WEB-INF/web.xml is no longer required.
+@WebServlet(name = "HelloAppEngine", value = "/hello")
+public class HelloAppEngine extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    String project = "YOUR_PROJECT_NAME";
+    String bucket = "gs://YOUR_BUCKET_NAME";
+
+    ArrayList<String> scopes = new ArrayList<String>();
+    scopes.add("https://www.googleapis.com/auth/cloud-platform");
+    final AppIdentityService appIdentity = AppIdentityServiceFactory.getAppIdentityService();
+    final AppIdentityService.GetAccessTokenResult accessToken = appIdentity.getAccessToken(scopes);
+
+    JSONObject jsonObj = null;
+    try {
+      JSONObject parameters = new JSONObject()
+        .put("datastoreReadGqlQuery", "SELECT * FROM Entries")
+        .put("datastoreReadProjectId", project)
+        .put("textWritePrefix", bucket + "/output/");
+      JSONObject environment = new JSONObject()
+        .put("tempLocation", bucket + "/tmp/")
+        .put("bypassTempDirValidation", false);
+      jsonObj = new JSONObject()
+        .put("jobName", "template-" + UUID.randomUUID().toString())
+        .put("parameters", parameters)
+        .put("environment", environment);
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+
+    URL url = new URL("https://dataflow.googleapis.com/v1b3/projects/"
+        + project + "/templates:launch?gcs_path="
+        + "gs://dataflow-templates/latest/Datastore_to_GCS_Text");
+    HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+    conn.setDoOutput(true);
+    conn.setRequestMethod("POST");
+    conn.setRequestProperty("Authorization", "Bearer " + accessToken.getAccessToken());
+    conn.setRequestProperty("Content-Type", "application/json");
+
+    OutputStreamWriter writer = new OutputStreamWriter(conn.getOutputStream());
+    jsonObj.write(writer);
+    writer.close();
+
+    int respCode = conn.getResponseCode();
+    if (respCode == HttpURLConnection.HTTP_OK) {
+      response.setContentType("application/json");
+      String line;
+      BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+      while ((line = reader.readLine()) != null) {
+        response.getWriter().println(line);
+      }
+      reader.close();
+
+    } else {
+      StringWriter w = new StringWriter();
+      IOUtils.copy(conn.getErrorStream(), w, "UTF-8");
+      response.getWriter().println(w.toString());
+    }
+  }
+
+  public static String getInfo() {
```

Comment:
Is this helpful?

Reply:
My goal was to minimize the delta between this example and the original HelloAppEngine sample. Removing this meant ripping a chunk of the tests out as well. I have no problem removing both for simplicity's sake now this this example is being distanced from the original.